### PR TITLE
Add set user functionality

### PR DESCRIPTION
**Pull request by tyanhan**

### CODE REVIEW

File: src/test/java/seedu/address/logic/commands/SetUserCommandTest.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    private Model model = new ModelManager(getTypicalAddressBook(), new UserPrefs());
+
+    @Test
+    public void execute_validIndexUnfilteredList_success() {
+        Person personToSetUser = model.getFilteredPersonList().get(INDEX_FIRST_PERSON.getZeroBased());
+        SetUserCommand setUserCommand = new SetUserCommand(INDEX_FIRST_PERSON);
+
+        String expectedMessage = String.format(SetUserCommand.MESSAGE_SET_USER_SUCCESS, personToSetUser.getName());
+
+        ModelManager expectedModel = new ModelManager(model.getAddressBook(), new UserPrefs());
```
Comment (vishandi) [inline src/test/java/seedu/address/logic/commands/SetUserCommandTest.java:36]: can move this one to line 28, so it's consistent with the other test files.

Comment (vishandi) [inline src/test/java/seedu/address/logic/commands/SetUserCommandTest.java:36]: can move this one to line 28, so it's consistent with the other test files.

File: src/test/java/seedu/address/logic/commands/SetUserCommandTest.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+ * Contains integration tests (interaction with the Model) and unit tests for
+ * {@code SetUserCommand}.
+ */
+public class SetUserCommandTest {
+
+    private Model model = new ModelManager(getTypicalAddressBook(), new UserPrefs());
+
+    @Test
+    public void execute_validIndexUnfilteredList_success() {
+        Person personToSetUser = model.getFilteredPersonList().get(INDEX_FIRST_PERSON.getZeroBased());
```
Comment (vishandi) [inline src/test/java/seedu/address/logic/commands/SetUserCommandTest.java:31]: I think this one can test INDEX_SECOND_PERSON instead?

File: src/test/java/seedu/address/logic/parser/person/SetUserCommandParserTest.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+import seedu.address.logic.commands.person.SetUserCommand;
+
+/**
+ * As we are only doing white-box testing, our test cases do not cover path variations
+ * outside of the SetUserCommand code. For example, inputs "1" and "1 abc" take the
+ * same path through the SetUserCommand, and therefore we test only one of them.
+ * The path variation for those two cases occur inside the ParserUtil, and
+ * therefore should be covered by the ParserUtilTest.
+ */
```
Comment (vishandi) [inline src/test/java/seedu/address/logic/parser/person/SetUserCommandParserTest.java:18]: is this one necessary? Cos if yes I need to add this to ViewScheduleCommandParserTest.java as well lol

File: src/test/java/seedu/address/logic/commands/SetUserCommandTest.java [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    public void execute_invalidIndexUnfilteredList_throwsCommandException() {
+        Index outOfBoundIndex = Index.fromOneBased(model.getFilteredPersonList().size() + 1);
+        SetUserCommand setUserCommand = new SetUserCommand(outOfBoundIndex);
+
+        assertCommandFailure(setUserCommand, model, Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+    }
+
+    @Test
+    public void execute_validIndexFilteredList_success() {
+        showPersonAtIndex(model, INDEX_FIRST_PERSON);
```
Comment (vishandi) [inline src/test/java/seedu/address/logic/commands/SetUserCommandTest.java:53]: same thing here, I think can try `showPersonAtIndex(model, INDEX_SECOND_PERSON);`

**Review by vishandi — CHANGES_REQUESTED** [review]
Looks good, only suggesting minor changes.

File: src/test/java/seedu/address/logic/parser/person/SetUserCommandParserTest.java (hunk 3 above)
Comment (tyanhan) [reply src/test/java/seedu/address/logic/parser/person/SetUserCommandParserTest.java:18]: To be honest I copy pasted this from `DeleteCommandParserTest`, it should not be necessary, I will remove.

File: src/test/java/seedu/address/logic/commands/SetUserCommandTest.java (hunk 4 above)
Comment (tyanhan) [reply src/test/java/seedu/address/logic/commands/SetUserCommandTest.java:53]: Yup sounds good!

**Pull request merged**